staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go

```diff
@@ -236,3 +236,42 @@ func (w WithMaxWaitRateLimiter) Forget(item interface{}) {
 func (w WithMaxWaitRateLimiter) NumRequeues(item interface{}) int {
 	return w.limiter.NumRequeues(item)
 }
+
+var _ RateLimiter = &FakeRateLimiter{}
+
+// FakeControllerRateLimiter is a constructor for a rate limiter for a workqueue.
+func FakeControllerRateLimiter(baseDelay, maxDelay time.Duration, limit int) RateLimiter {
+	return NewMaxOfRateLimiter(
+		NewFakeRateLimiter(baseDelay, maxDelay, limit),
+	)
+}
+
+func NewFakeRateLimiter(baseDelay time.Duration, maxDelay time.Duration, limit int) RateLimiter {
+	return &FakeRateLimiter{
+		baseDelay: baseDelay,
+		maxDelay:  maxDelay,
+		limit:     limit,
+	}
+}
+
+type FakeRateLimiter struct {
```

Comment:
How about `StepRateLimiter` since the output of this rate limiter is a step function? A more descriptive name gives us space to add other types of fake rate limiters in the future.

Reply:
Renamed to `StepRateLimiter`